File: Allocore-frontend/src/types/costCenter.ts
export interface CostCenterDto {
  id: string;
  companyId: string;
  code: string;
  name: string;
  description: string | null;
  isActive: boolean;
  employeeCount: number;
  createdAt: string;
  updatedAt: string | null;
}

export interface CostCenterListItemDto {
  id: string;
  code: string;
  name: string;
  isActive: boolean;
  employeeCount: number;
}

export interface CreateCostCenterRequest {
  code: string;
  name: string;
  description?: string | null;
}

export interface UpdateCostCenterRequest {
  code: string;
  name: string;
  description?: string | null;
}

File: Allocore-backend/src/Allocore.Domain/Entities/CostCenters/CostCenter.cs
namespace Allocore.Domain.Entities.CostCenters;

using Allocore.Domain.Common;

public class CostCenter : Entity
{
    public Guid CompanyId { get; private set; }
    public string Code { get; private set; } = string.Empty;
    public string Name { get; private set; } = string.Empty;
    public string? Description { get; private set; }
    public bool IsActive { get; private set; } = true;

    private CostCenter() { } // EF Core

    public static CostCenter Create(
        Guid companyId,
        string code,
        string name,
        string? description = null)
    {
        return new CostCenter
        {
            CompanyId = companyId,
            Code = code.Trim().ToUpperInvariant(),
            Name = name.Trim(),
            Description = description?.Trim(),
            IsActive = true
        };
    }

    public void Update(string code, string name, string? description)
    {
        Code = code.Trim().ToUpperInvariant();
        Name = name.Trim();
        Description = description?.Trim();
        UpdatedAt = DateTime.UtcNow;
    }

    public void Deactivate()
    {
        IsActive = false;
        UpdatedAt = DateTime.UtcNow;
    }

    public void Activate()
    {
        IsActive = true;
        UpdatedAt = DateTime.UtcNow;
    }
}
